### PR TITLE
Packaging for release 14.4.1. Adding debug logs to investigate an authentication issue.

### PR DESCRIPTION
**Pull request by lstp**
We are trying to debug an issue with authentication in POS Channel and need additional logs to do so. See:
https://github.com/Shopify/shopify_app/issues/1073

Before submitting the PR, please consider if any of the following are needed:

- [x] Update `CHANGELOG.md` if the changes would impact users
- [x] Update `README.md`, if appropriate.
- [x] Update any relevant pages in `docs/`, if necessary
- [x] For security fixes, the [Disclosure Policy](https://github.com/Shopify/shopify_app/blob/master/SECURITY.md#disclosure-policy) must be followed.

### CODE REVIEW

File: app/controllers/shopify_app/callback_controller.rb [hunk 1]
```diff
@@ -11,17 +11,21 @@ def callback
...
+        Rails.logger.debug("[ShopifyApp::CallbackController] Invalid JWT auth detected.")
         return respond_with_error
       end
 
       if jwt_request?
+        Rails.logger.debug("[ShopifyApp::CallbackController] JWT request detected. Setting shopify session...")
         set_shopify_session
         head(:ok)
       else
+        Rails.logger.debug("[ShopifyApp::CallbackController] Not a JWT request. Reseting session options...")
```
Comment (minichate) [inline app/controllers/shopify_app/callback_controller.rb:23]: nitpick: Should be `Resetting`

**Review by rezaansyed — APPROVED** [review]
LGTM!

**Review by minichate — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,7 @@
+14.4.1
```
Comment (andyw8) [inline CHANGELOG.md:1]: @lstp Once we solve the issue, are we planning to keep the extra logging or remove it?

(This is a public gem)

Comment (lstp) [reply CHANGELOG.md:1]: I think it's useful to keep them but no objections to remove them if required.

**Review by paulomarg — APPROVED** [review]
LGTM!

I am slightly concerned that we'll be logging quite a few things for partners after they start running this version, but these *are* all debug logs, so I don't see it as a problem.

Nitpick: usually we separate the 'real' PRs from the release so we can follow this history from the commits. But these changes are pretty minor!

**Review by andyw8 — APPROVED** [review]

**Comment by andyw8** [comment]
> Nitpick: usually we separate the 'real' PRs from the release so we can follow this history from the commits. But these changes are pretty minor!

Yeah, we should update the release docs to make that clearer.

@lstp Can we update the PR title to make it clearer that this is adding the logging as well as updating the packaging?

**Comment by minichate** [comment]
@andyw8 We're going to tag this as `v14.4.1` -- planning on:
- merging into master
- tagging v14.4.1 from _this branch_ (NOT master!)
- pushing the tag
- ShipIt builds the tag/gem

Make sense? We don't want to ship the other commits that are in master now

**Pull request merged**